src/routing/Routing_Py_Adapter.cpp

```diff
@@ -13,14 +12,15 @@ using namespace routing_py_adapter;
  * Parameterized constructor Routing_Py_Adapter with the flow_vector
  *
  * @param t_route_connection_path
+ * @param t_route_config_file_with_path
  * @param input_path
- * @param catchment_subset_ids
  * @param number_of_timesteps
  * @param delta_time
  * @param flow_vector
  */
-Routing_Py_Adapter::Routing_Py_Adapter(std::string t_route_connection_path, std::string input_path,
-                                       const std::vector<std::string> &catchment_subset_ids,
+Routing_Py_Adapter::Routing_Py_Adapter(std::string t_route_connection_path, 
+                                       std::string t_route_config_file_with_path,
+                                       std::string input_path,
```

Comment:
IIRC this may have been a constructor signature used to consider "integrated" routing, i.e. running a routing step after each catchment update...

I don't think it would make sense to recreate the adapter for each loop instance though, so I don't see this as being a useful/needed constructor.

Thinking about it a bit, we probably want to decouple the construction from the run, then we could provide functions/overloads for each case without the overhead.  I will refactor the construction and this class to reflect that.

Reply:
Refactored in 9cf4426d4793e7241315e354126ba516b85bcdbb